js/layout-javascript/agenda-code.js

```diff
@@ -856,6 +856,10 @@ DynamicList.prototype.initialize = function() {
         uuid: _this.data.uuid,
         container: _this.$container,
         records: records
+      }).then(function() {
+        return Fliplet.User.getCachedSession();
+      }).then(function(session) {
+        _this.Utils.Records.setSource(session, _this.data);
```

Comment:
`_this.Utils.Records.setSource()` returns a promise, which suggests in most cases that we should wait for the promises to resolve before continuing.

```suggestion
        return _this.Utils.Records.setSource(session, _this.data);
```